ngraph/test/util/onnx_test_util.cpp

```diff
@@ -0,0 +1,311 @@
+//*****************************************************************************
+// Copyright 2017-2021 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include <exception>
+#include <fstream>
+#include <google/protobuf/io/zero_copy_stream_impl.h>
+#include <google/protobuf/text_format.h>
+#include <onnx/onnx_pb.h>
+
+#include "util/onnx_test_util.hpp"
+
+namespace
+{
+    ONNX_NAMESPACE::ModelProto parse_from_file(const std::string& file_path)
+    {
+        std::ifstream file_stream{file_path, std::ios::in | std::ios::binary};
+
+        if (!file_stream.is_open())
+        {
+            throw std::runtime_error("Could not open the file: " + file_path);
+        };
+
+        if (!file_stream.good())
+        {
+            file_stream.clear();
+            file_stream.seekg(0);
+            if (!file_stream.good())
+            {
+                throw std::runtime_error("Invalid state of model's input stream.");
+            }
+        }
+
+        ONNX_NAMESPACE::ModelProto model_proto;
+        if (!model_proto.ParseFromIstream(&file_stream))
+        {
+#ifdef NGRAPH_USE_PROTOBUF_LITE
+            throw ngraph_error("Could not import the model");
+#else
+            // Rewind to the beginning and clear stream state.
+            file_stream.clear();
+            file_stream.seekg(0);
+            google::protobuf::io::IstreamInputStream iistream(&file_stream);
+            // Try parsing input as a prototxt message
+            if (!google::protobuf::TextFormat::Parse(&iistream, &model_proto))
+            {
+                throw std::runtime_error("Could not import the prototxt model");
+            }
+#endif
+        }
+
+        return model_proto;
+    }
+
+    const std::pair<bool, std::string> COMPARISON_OK{true, {}};
```

Comment:
**Personally** I dislike this approach. 
My suggestion:
```
struct Result {
    bool valid{true};
    std::string message;
    static Result ok() { return {}; }
    static Result error(std::message m){return Resutl{false, std::move(m);}
}
```
Then you can check:
```
const auto result = func();
if ( !result.valid ) { std::cout << resutl.message << std::endl; }
```
I dream to have such class in some general scope.

Reply:
Good idea, done.